### DISCUSSION TITLE
Releationship between WatchDogTermination and OOM

### DISCUSSION THREAD

**Discussion by echives** [getsentry/sentry-cocoa#3229] [Q&A]
Our team wants to monitor OOM recently, and i wonder if we can use Sentry to achieve that. After my test with Sentry WatchDogTermination, i have the following questions, can someone help me out🙏🏻
1. The difference between WatchDogTermination and OOM
2. I wrote some OOM codes and found that OOM does not necessarily trigger the WatchDogTermination report. What is the reason? test code like this `DispatchQueue.global().async {
            var a = [Int]()
            while true {
                a.append(1)
            }
        }`

thanks

**Comment by brustolin** [comment]
Hello @echives, [here](https://docs.sentry.io/platforms/apple/guides/ios/configuration/watchdog-terminations/) are the docs regarding WatchDogTermination (WDT).

1.  In summary every OOM should trigger a WDT, but not every WDT is an OOM. 
Right now, the only type of WDT we try do detect is OOM.

2.  Did you run your code on a simulator or a device? Did you re-open the app after it was terminated in order to send the WDT report?
